api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -507,4 +538,49 @@ public String toString() {
                 .map(this::identifierFieldToString)
                 .collect(Collectors.toList())));
   }
+
+  /**
+   * All ids of metadata fields are reassigned.
+   *
+   * @return map of original to reassigned field ids of metadata fields
+   */
+  public Map<Integer, Integer> idsToReassigned() {
+    return idsToReassigned != null ? idsToReassigned : Maps.newHashMap();
```

Comment:
Default should probably be an immutable empty map instead of a new object

Reply:
Fixed.